Core/Interfaces/IRepository.cs

```diff
@@ -0,0 +1,15 @@
+﻿using LearnNet_CatalogService.Data.Common;
+using System.Linq.Expressions;
+
+namespace LearnNet_CatalogService.Core.Interfaces
+{
+    public interface IRepository<T> where T : BaseEntity
+    {
+        public Task<IQueryable<T>> GetAll();
```

Comment:
why does this method return IQueryable? The data should be aggregated by the repository

Reply:
I think that data should be aggregated on service level, just in cause of some additional aggregation etc.

Reply:
I disagree, please read: https://stackoverflow.com/questions/33755499/entity-framework-repository-pattern-why-not-return-iqueryable
and other similar articles.
Repository class should encapsulate all aggregation logic and return aggregated data to the service level